apps/picsa-apps/dashboard/src/app/modules/crop-information/crop-information.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@angular/core';
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import { Database } from '@picsa/server-types';
+import { PicsaAsyncService } from '@picsa/shared/services/asyncService.service';
+import { SupabaseService } from '@picsa/shared/services/core/supabase';
+import { IStorageEntry } from '@picsa/shared/services/core/supabase/services/supabase-storage.service';
+
+export type ICropInformationRow = Database['public']['Tables']['crop_data']['Row'];
+
+export interface IResourceStorageEntry extends IStorageEntry {
+  /** Url generated when upload to public bucket (will always be populated, even if bucket not public) */
+  publicUrl: string;
+}
+
+@Injectable({ providedIn: 'root' })
+export class CropProbabilityDashboardService extends PicsaAsyncService {
+  public cropProbabilities: ICropInformationRow[] = [];
+
+  public get table() {
+    return this.supabaseService.db.table('station_crop_information');
+  }
+
+  constructor(private supabaseService: SupabaseService) {
+    super();
+  }
+
+  public override async init() {
+    await this.supabaseService.ready();
+    await this.listCropProbabilities();
+  }
+
+  public async listCropProbabilities() {
+    const { data, error } = await this.supabaseService.db
+      .table('station_crop_information')
+      .select<'*', ICropInformationRow>('*');
+    if (error) {
+      throw error;
+    }
+    this.cropProbabilities = data || [];
+  }
+
+  public async addCropProbability(cropProbability: Partial<ICropInformationRow>): Promise<string> {
```

Comment:
nit(non-blocking) - if you want to be more strict with type definitions the generated types also include separately the data requirements for inserting rows `Database['public']['Tables']['crop_data']['Insert']` - although I don't think a huge deal as we'll likely manage data validation within the form so doubt it would pick up many type errors.